### PR TITLE
[GCP FileShare Support] Fix for Invalid memory address or nil pointer dereference for GCP Fileshare Creation

### LINKED ISSUE #1104: [GCP FileShare Support] Invalid memory address or nil pointer dereference for GCP Fileshare Creation
**Issue/Feature Description:** GCP Fileshare Creation is failing with the following reason:

Invalid memory address or nil pointer dereference

image

Error logs:
```
[2020-09-27T16:22:03+05:30] [ERROR] [/root/gopath/src/github.com/opensds/multi-cloud/file/pkg/service/service.go] [service.(*fileService).CreateFileShare():164] [PID:1] Received error in creating file shares at backend createInstance operation failed: googleapi: Error 429: Quota limit 'StandardStorageGbPerRegion' has been exceeded. Limit: 2048 in region asia-east1., rateLimitExceeded
2020-09-27 16:22:03  file=grpc.go:372 level=error panic recovered: runtime error: invalid memory address or nil pointer dereference
2020-09-27 16:22:03  file=grpc.go:373 level=error goroutine 1759 [running]:
runtime/debug.Stack(0xc0000be480, 0xc0009a5a02, 0xc0009a5ae0)
        /usr/local/go/src/runtime/debug/stack.go:24 +0x9d
github.com/micro/go-micro/v2/server/grpc.(*grpcServer).processRequest.func1.1(0xc00091b250)
        /root/gopath/pkg/mod/github.com/micro/go-micro/v2@v2.6.0/server/grpc/grpc.go:373 +0x2cf
panic(0x1838560, 0x2c432c0)
        /usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/opensds/multi-cloud/file/pkg/service.(*fileService).CreateFileShare(0xc0007dcde0, 0x1ef0b00, 0xc000ca82d0, 0xc000ca80f0, 0xc000ca8180, 0x4d00ec, 0x1a05980)
        /root/gopath/src/github.com/opensds/multi-cloud/file/pkg/service/service.go:165 +0x380
github.com/opensds/multi-cloud/file/proto.(*fileHandler).CreateFileShare(0xc0004fda20, 0x1ef0b00, 0xc000ca82d0, 0xc000ca80f0, 0xc000ca8180, 0x7fc5a1984d98, 0x0)
        /root/gopath/src/github.com/opensds/multi-cloud/file/proto/file.pb.micro.go:153 +0x5b
reflect.Value.call(0xc000146180, 0xc00021a310, 0x13, 0x1b16bec, 0x4, 0xc00091b1b0, 0x4, 0x4, 0x16, 0x19ffe80, ...)
        /usr/local/go/src/reflect/value.go:460 +0x5f6
reflect.Value.Call(0xc000146180, 0xc00021a310, 0x13, 0xc00091b1b0, 0x4, 0x4, 0x203000, 0xc00091b1b0, 0x40c1d6)
        /usr/local/go/src/reflect/value.go:321 +0xb4
github.com/micro/go-micro/v2/server/grpc.(*grpcServer).processRequest.func1(0x1ef0b00, 0xc000ca82d0, 0x1f06f80, 0xc00026b700, 0x1a05980, 0xc000ca8180, 0x0, 0x0)
        /root/gopath/pkg/mod/github.com/micro/go-micro/v2@v2.6.0/server/grpc/grpc.go:378 +0x303
github.com/micro/go-micro/v2/util/wrapper.AuthHandler.func1.1(0x1ef0b00, 0xc000ca8270, 0x1f06f80, 0xc00026b700, 0x1a05980, 0xc000ca8180, 0xc000ca8270, 0xc0001077a0)
        /root/gopath/pkg/mod/github.com/micro/go-micro/v2@v2.6.0/util/wrapper/wrapper.go:282 +0x5e3
github.com/micro/go-micro/v2/util/wrapper.TraceHandler.func1.1(0x1ef0a40, 0xc0009ded40, 0x1f06f80, 0xc00026b700, 0x1a05980, 0xc000ca8180, 0x16, 0x1a05980)
        /root/gopath/pkg/mod/github.com/micro/go-micro/v2@v2.6.0/util/wrapper/wrapper.go:122 +0x1c6
github.com/micro/go-micro/v2/util/wrapper.HandlerStats.func1.1(0x1ef0a40, 0xc0009ded40, 0x1f06f80, 0xc00026b700, 0x1a05980, 0xc000ca8180, 0x14, 0x0)
        /root/gopath/pkg/mod/github.com/micro/go-micro/v2@v2.6.0/util/wrapper/wrapper.go:67 +0x83
github.com/micro/go-micro/v2/server/grpc.(*grpcServer).processRequest(0xc000087800, 0x1efc8a0, 0xc00030ca80, 0xc00060f500, 0xc000632000, 0xc0006d8b00, 0x16, 0x1ef0a40, 0xc0009ded40, 0x6d4a9cef17fafc18, ...)
        /root/gopath/pkg/mod/github.com/micro/go-micro/v2@v2.6.0/server/grpc/grpc.go:395 +0x7ee
github.com/micro/go-micro/v2/server/grpc.(*grpcServer).handler(0xc000087800, 0x0, 0x0, 0x1efc8a0, 0xc00030ca80, 0x0, 0x0)
        /root/gopath/pkg/mod/github.com/micro/go-micro/v2@v2.6.0/server/grpc/grpc.go:314 +0x77f
google.golang.org/grpc.(*Server).processStreamingRPC(0xc00021c300, 0x1f06fe0, 0xc000a8e780, 0xc0005fcb00, 0x0, 0xc0007dcd00, 0x0, 0x0, 0x0)
        /root/gopath/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:1237 +0xb42
google.golang.org/grpc.(*Server).handleStream(0xc00021c300, 0x1f06fe0, 0xc000a8e780, 0xc0005fcb00, 0x0)
        /root/gopath/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:1323 +0xc0a
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000a02210, 0xc00021c300, 0x1f06fe0, 0xc000a8e780, 0xc0005fcb00)
        /root/gopath/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:722 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /root/gopath/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:720 +0xa1
```
**Why this issue to fixed / feature is needed(give scenarios or use cases):** To Support GCP FileShare Creation with proper Error Handling

**How to reproduce, in case of a bug:**

1. Create GCP Backend
2. Create GCP File Share with 70258793014886 Bytes Size

**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**

### PR DESCRIPTION
**Pull request by himanshuvar**
**What type of PR is this?**
> /kind bug fix

**What this PR does / why we need it**: This PR will fix for nil pointer derefence for FileShare resouces in the service layer 

**Which issue(s) this PR fixes**:

Fixes #1104 

**Test Report Added?**:
> /kind TESTED

**Test Report**:

image

**Special notes for your reviewer**:

### CODE REVIEW

File: openapi-spec/swagger.yaml [hunk 1]
```diff
@@ -817,8 +817,8 @@ paths:
       responses:
         '200':
           description: >-
-           Get storage classes succesfully. There is a map between storage class name
-           and stroage tier, as default the storage classes are compatible with aws s3.
+            Get storage classes succesfully. There is a map between storage class name
+            and stroage tier, as default the storage classes are compatible with aws s3.
```
Comment (kumarashit) [inline openapi-spec/swagger.yaml:821]: Typo: strogae -> storage

File: openapi-spec/swagger.yaml [hunk 2]
```diff
@@ -817,8 +817,8 @@ paths:
       responses:
         '200':
           description: >-
-           Get storage classes succesfully. There is a map between storage class name
```
Comment (kumarashit) [inline openapi-spec/swagger.yaml:820]: Typo: succesfully -> successfully

Comment (himanshuvar) [reply openapi-spec/swagger.yaml:820]: Done, Thanks

File: openapi-spec/swagger.yaml (hunk 1 above)
Comment (himanshuvar) [reply openapi-spec/swagger.yaml]: Done, thanks

**Review by kumarashit — APPROVED** [review]
LGTM

**Review by skdwriting — APPROVED** [review]
LGTM

**Review by skdwriting — APPROVED** [review]
LGTM

**Pull request merged**